portfolio/src/main/java/com/google/sps/servlets/SleepDataServlet.java

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+
+@WebServlet("/sleep-data")
+public class SleepDataServlet extends HttpServlet {
+  static final int SIZE = 5;
+  static final String FILENAME = "/WEB-INF/sleep_data.csv";
+
+  private LinkedHashMap<String, Double[]> sleepData = new LinkedHashMap<>();
+  
+
+  /** 
+   * Reads in csv file and parses the data into the sleepData HashMap. Calls 
+   * the movingAverage function to add the moving average data as addition 
+   * elements in the array (value of the HashMap).
+   */
+  @Override
+  public void init() {
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream(FILENAME));
+    while (scanner.hasNextLine()) {
+        String line = scanner.nextLine();
+        String[] cells = line.split(",");
+
+        String timestamp = cells[0];
+        int tindex = timestamp.indexOf("T"); 
+        String date_string = timestamp.substring(0 , tindex);
```

Comment:
This makes assumption about the format of the timestamp. Can you explain what is going on and also show an example?

Reply:
I have the raw data in a csv file with the timestamp, sleep score, duration score, and deep sleep minutes. All the timestamps are of the form 2020-06-25T07:26:30Z, so I split it at "T" and take the first split. I added a comment in the java file explaining it.